test-plans/Makefile

```diff
@@ -0,0 +1,10 @@
+all: ping-image.tar
+
+ping-image.tar: Dockerfile Cargo.toml $(wildcard src/*)
+	cd .. && docker build -t rust-libp2p-head -f test-plans/Dockerfile .
```

Comment:
This is just an intuition but I think this will send the entire directory to the Docker daemon as "context" so the actual build takes long because it copies the entire target directory to somewhere but it only ends up picking one binary out of it.

It might be worthwhile to move the binary out of the target dir into the `test-plans` dir and run `docker build` from within there instead (and remove the `cd ..`)

Reply:
yup, agreed and updated. Thanks Thomas